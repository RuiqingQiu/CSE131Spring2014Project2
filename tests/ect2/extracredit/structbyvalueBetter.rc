const int c = 10;

structdef S { int [c] a; };

function : void foo (S x, S y, S z){

   int i = 0;

   while (i < c){
      y.a[i] = (z.a[i] = (x.a[i] = i) + 10000) + 100;
   i++;}
   
   i = 0;
   while (i < c){
      cout << x.a[i] << "." << y.a[i] << "." <<  z.a[i++] << endl;
   }
}

function : int main(){  
   S x, y, z;
   
   int i = 0; 
   
   foo(x,y,z);
   
   while (i < c){
      x.a[i] = (y.a[i] = (z.a[i] = i) + 1) + 1;
   i++;}
   
   
   i = 0;
   while (i < c){
      cout << x.a[i] << "." << y.a[i] << "." <<  z.a[i++] << endl;
   }

   return 0;
}
