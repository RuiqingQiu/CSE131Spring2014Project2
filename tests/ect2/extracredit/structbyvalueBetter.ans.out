0.10100.10000
1.10101.10001
2.10102.10002
3.10103.10003
4.10104.10004
5.10105.10005
6.10106.10006
7.10107.10007
8.10108.10008
9.10109.10009
2.1.0
3.2.1
4.3.2
5.4.3
6.5.4
7.6.5
8.7.6
9.8.7
10.9.8
11.10.9
